Lake/Config/Module.lean

```diff
@@ -20,6 +20,9 @@ structure Module where
   keyName : Name := name
   deriving Inhabited
 
+instance : Hashable Module where hash m := hash m.name
+instance : BEq Module where beq m n := m.name == n.name
```

Comment:
```suggestion
instance : Hashable Module where hash m := hash m.keyName
instance : BEq Module where beq m n := m.keyName == n.keyName
```

While two  different modules can share the same name in a build (e.g., `Main` modules for executables of different packages), key names are unique and thus should be used here.